### PR TITLE
Kenny/sm gatk

### PR DESCRIPTION
**Pull request by kennyworkman**
* Replace skipped rules with Ellipsis. Supports rules nested in conditionals where previously an empty block was produced.
* Patched parser to generate compiled code for `workflow.include` calls https://github.com/snakemake/snakemake/pull/2469
* Detect use of conda keyword and install in image. This effectively supports `wrapper`/`conda` keywords.
* Support for `multi-ext` keyword

### CODE REVIEW

File: latch_cli/snakemake/single_task_snakemake.py [hunk 1]
```diff
@@ -28,6 +29,11 @@ def eprint(x: str) -> None:
     print(x, file=sys.stderr)
 
 
+print_compilation_env = os.environ["LATCH_PRINT_COMPILATION"]
+print_compilation = False
+if print_compilation_env == "1":
+    print_compilation = True
```
Comment (ayushkamat) [inline latch_cli/snakemake/single_task_snakemake.py:35]:
```suggestion
print_compilation = False
if os.environ.get("LATCH_PRINT_COMPILATION") == "1":
    print_compilation = True
```

File: latch_cli/snakemake/single_task_snakemake.py [hunk 2]
```diff
@@ -154,10 +160,26 @@ def emit_overrides(self, token):
...
+        and xs["positional"][0].get("flags")
+        and "multiext" in xs["positional"][0].get("flags")
+    ):
+        eprint("inside")
+        quote = "'"
+        positional_data = (
+            (
+                f"multiext('{xs[0]['value']}',{','.join([quote + x['value'].split('.')[-1] + quote for x in xs])})"
+            ),
+        )
```
Comment (ayushkamat) [inline latch_cli/snakemake/single_task_snakemake.py:174]:
```suggestion
        positional_data = (
            (
                f"multiext({repr(xs[0]['value'])},{','.join([repr(x['value'].split('.')[-1]) for x in xs])})"
            ),
        )
```

Comment (ayushkamat) [reply latch_cli/snakemake/single_task_snakemake.py:174]: also shouldnt it be `xs["positional"]` not just `xs`

File: latch_cli/snakemake/single_task_snakemake.py [hunk 3]
```diff
@@ -188,17 +210,33 @@ def skipping_block_content(self, token):
...
         if self.rulename not in rules:
+            # Rules can be nested in conditional statements:
+            #
+            # if (<condition>):
+            #   rule A:
+            #       <stuff>
+            #
+            # We want correct python code if we remove them.
+            yield "..."
             return
```
Comment (ayushkamat) [inline latch_cli/snakemake/single_task_snakemake.py:240]: del the return

File: latch_cli/snakemake/single_task_snakemake.py (hunk 1 above)
Comment (ayushkamat) [reply latch_cli/snakemake/single_task_snakemake.py:35]: actually why get this from the env in the first place - seems like an unnecessary switch considering that the env variable is always being set anyway

just have it print-compilation always

**Review by ayushkamat — CHANGES_REQUESTED** [review]
@maximsmol should also look

File: latch_cli/snakemake/single_task_snakemake.py [hunk 4]
```diff
@@ -154,10 +159,26 @@ def emit_overrides(self, token):
     else:
         raise ValueError(f"tried to emit overrides for unknown state: {type(self)}")
 
-    positional_data = (render_annotated_str_list(x) for x in xs["positional"])
-    keyword_data = (
-        f"{k}={render_annotated_str_list(v)}" for k, v in xs["keyword"].items()
-    )
+    if (
+        isinstance(self, Output)
+        and xs["positional"][0].get("flags")
```
Comment (maximsmol) [inline latch_cli/snakemake/single_task_snakemake.py:164]:
```suggestion
        and xs["positional"][0].get("flags") is not None
```

File: latch_cli/snakemake/single_task_snakemake.py [hunk 5]
```diff
@@ -154,10 +159,26 @@ def emit_overrides(self, token):
...
+        and "multiext" in xs["positional"][0].get("flags")
+    ):
+        eprint("inside")
```
Comment (maximsmol) [inline latch_cli/snakemake/single_task_snakemake.py:167]: del?

File: latch_cli/snakemake/single_task_snakemake.py [hunk 6]
```diff
@@ -154,10 +159,26 @@ def emit_overrides(self, token):
...
+        quote = "'"
+        positional_data = (
+            (
+                f"multiext('{xs[0]['value']}',{','.join([quote + x['value'].split('.')[-1] + quote for x in xs])})"
```
Comment (maximsmol) [inline latch_cli/snakemake/single_task_snakemake.py:171]:
```suggestion
                f"multiext({repr(xs[0]['value'])},{','.join([repr(x['value'].split('.')[-1]) for x in xs])})"
```

File: latch_cli/snakemake/single_task_snakemake.py (hunk 1 above)
Comment (kennyworkman) [reply latch_cli/snakemake/single_task_snakemake.py]: its not always being set, its only being set when we want to print the compilation

File: latch_cli/snakemake/single_task_snakemake.py (hunk 2 above)
Comment (kennyworkman) [reply latch_cli/snakemake/single_task_snakemake.py:173]: yes

File: latch_cli/snakemake/single_task_snakemake.py (hunk 3 above)
Comment (kennyworkman) [reply latch_cli/snakemake/single_task_snakemake.py:239]: we need the return

**Review by maximsmol — CHANGES_REQUESTED** [review]

**Pull request merged**